### PR TITLE
docs(ce/2.1) clone UPGRADE.md from the kong/kong repo

### PR DESCRIPTION
**Pull request by hishamhm**

### CODE REVIEW

File: app/2.1.x/upgrading.md [hunk 1]
```diff
@@ -2,21 +2,202 @@
 title: Upgrade guide
```
Comment (lena-larionova) [inline app/2.1.x/upgrading.md:2]:
```suggestion
title: Upgrade guide
skip_read_time: true
```
Right now the estimated reading time is 102 minutes, no need to scare anyone.

File: app/2.1.x/upgrading.md [hunk 2]
```diff
@@ -2,21 +2,202 @@
...
+starts new workers, which take over from old workers before those old workers
+are terminated. In this way, Kong will serve new requests via the new
+configuration, without dropping existing in-flight connections.
+
+
+## Upgrade to `2.1.0`
+
+Kong adheres to [semantic versioning](https://semver.org/), which makes a
+distinction between "major", "minor", and "patch" versions. The upgrade path
+will be different on which previous version from which you are migrating.
```
Comment (tlblessing) [inline app/2.1.x/upgrading.md:48]:
```suggestion
will be different depending on which previous version from which you are migrating.
```

File: app/2.1.x/upgrading.md [hunk 3]
```diff
@@ -2,21 +2,202 @@
...
+5. Gradually divert traffic away from your old nodes, and into
+   your 2.1.0 cluster. Monitor your traffic to make sure everything
+   is going smoothly.
+6. When your traffic is fully migrated to the 2.1.0 cluster,
+   decommission your old nodes.
+
+##### Installing 2.1.0 on a Fresh Datastore
+
+The following commands should be used to prepare a new 2.1.0 cluster from a
+fresh datastore. By default the `kong` CLI tool will load the configuration
```
Comment (tlblessing) [inline app/2.1.x/upgrading.md:187]:
```suggestion
fresh datastore. By default, the `kong` CLI tool loads the configuration
```

File: app/2.1.x/upgrading.md [hunk 4]
```diff
@@ -2,21 +2,202 @@
...
+please migrate to 1.5.0 first.
+
+The steps for upgrading from 0.14.1 to 1.5.0 are the same as upgrading
+from 0.14.1 to Kong 1.0. Please follow the steps described in the
+"Migration Steps from 0.14" in the [Suggested Upgrade Path for Kong
+1.0](#kong-1-0-upgrade-path), with the addition of the `kong
+migrations migrate-apis` command, which you can use to migrate legacy
+`apis` configurations.
+
+Once you migrated to 1.5.0, you can follow the instructions in the section
```
Comment (tlblessing) [inline app/2.1.x/upgrading.md:121]:
```suggestion
After you have migrated to 1.5.0, follow the instructions in the section
```

File: app/2.1.x/upgrading.md [hunk 5]
```diff
@@ -2,21 +2,202 @@
...
+   your 2.1.0 cluster. Monitor your traffic to make sure everything
+   is going smoothly.
+6. When your traffic is fully migrated to the 2.1.0 cluster,
+   decommission your old nodes.
+
+##### Installing 2.1.0 on a Fresh Datastore
+
+The following commands should be used to prepare a new 2.1.0 cluster from a
+fresh datastore. By default the `kong` CLI tool will load the configuration
+from `/etc/kong/kong.conf`, but you can optionally use the flag `-c` to
```
Comment (tlblessing) [inline app/2.1.x/upgrading.md:188]:
```suggestion
from `/etc/kong/kong.conf`, but you can optionally use the `-c` flag to
```

File: app/2.1.x/upgrading.md [hunk 6]
```diff
@@ -2,21 +2,202 @@
...
+indicate the path to your configuration file:
+
+```
+$ kong migrations bootstrap [-c /path/to/your/kong.conf]
```
Comment (tlblessing) [inline app/2.1.x/upgrading.md:192]: Elsewhere, such as line 754, the format [-c config] is used. Other spots, such as line 27, it is  [-c configuration_file].
We should be consistent. I'm basing the KE migrations doc off of some portions of CE migrations doc.

File: app/2.1.x/upgrading.md [hunk 7]
```diff
@@ -2,21 +2,202 @@
...
+   decommission your old nodes.
+
+##### Installing 2.1.0 on a Fresh Datastore
+
+The following commands should be used to prepare a new 2.1.0 cluster from a
+fresh datastore. By default the `kong` CLI tool will load the configuration
+from `/etc/kong/kong.conf`, but you can optionally use the flag `-c` to
+indicate the path to your configuration file:
+
+```
```
Comment (tlblessing) [inline app/2.1.x/upgrading.md:191]:
```suggestion
```bash
```

or shell. global review comment for syntax highlighting.

File: app/2.1.x/upgrading.md (hunk 5 above)
Comment (tlblessing) [inline app/2.1.x/upgrading.md:188]: is the default dir the same for KE (EE)? I'm plagiarizing some of your verbiage.

File: app/2.1.x/upgrading.md [hunk 8]
```diff
@@ -2,21 +2,202 @@
...
-This guide will inform you about breaking changes you should be aware of
-when upgrading, as well as take you through the correct sequence of steps
-in order to obtain a **no-downtime** migration in different upgrade
-scenarios.
 
 ## Upgrade to `2.0.0`
 
 Kong adheres to [semantic versioning](https://semver.org/), which makes a
-distinction between "major", "minor" and "patch" versions. The upgrade path
+distinction between "major", "minor", and "patch" versions. The upgrade path
```
Comment (tlblessing) [inline app/2.1.x/upgrading.md:200]:
```suggestion
distinction between major, minor, and patch versions. The upgrade path
```

don't need quotes. global comment for similar sections. We can set up includes to share content in the future for better consistency and less maintenance.

File: app/2.1.x/upgrading.md [hunk 9]
```diff
@@ -486,3 +667,2607 @@ fresh datastore:
...
+
+</p>
+</details>
+
+#### 2. Suggested Upgrade Path
+
+##### Upgrade from `0.x` to `1.3`
+
+The lowest version that Kong 1.3 supports migrating from is 0.14.1. if you
+are migrating from a previous 0.x release, please migrate to 0.14.1 first.
```
Comment (tlblessing) [inline app/2.1.x/upgrading.md:1048]: Some style guides say to avoid "please". _Maybe_ use for contacting support.

**Pull request merged**